src/Utils/kernel_launching.jl

```diff
@@ -6,15 +6,31 @@ using Oceananigans.Architectures
 using Oceananigans.Grids
 using Oceananigans.Grids: AbstractGrid
 
-"""Parameters for kernel launch, containing kernel size (`S`) and kernel offsets (`O`)"""
 struct KernelParameters{S, O} end
 
+"""
+    KernelParameters(size, offsets)
+
+Return parameters for kernel launching and execution that define
+(i) a tuple that defines the `size` of the kernel being launched and
+(ii) a tuple of `offsets` that dictates where 
+
+Example
+=======
+
+```julia
+size = (8, 6, 4)
+offsets = (0, 1, 2)
+kp = KernelParameters(size, offsets)
+
+# Launch a kernel with indices that range from i=1:8, j=2:7, k=3:6,
+# where i, j, k are the first, second, and third index, respectively:
+launch!(arch, grid, kp, kernel!, kernel_args...)
+```
+"""
 KernelParameters(size, offsets) = KernelParameters{size, offsets}()
```

Comment:
@simone-silvestri please advice if this docstring makes sense

Reply:
makes sense